### PR TITLE
abinit: add version 10.0.7

### PR DESCRIPTION
**Pull request by downloadico**
Update abinit version to 10.0.7

### CODE REVIEW

**Review by tldahlgren — APPROVED** [review]
Confirmed the sha256.

File: var/spack/repos/builtin/packages/abinit/package.py [hunk 1]
```diff
@@ -126,7 +127,7 @@ class Abinit(AutotoolsPackage):
     )
 
     patch("rm_march_settings.patch", when="@:8")
-    patch("rm_march_settings_v9.patch", when="@9:")
+    patch("rm_march_settings_v9.patch", when="@9:9.99")
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/abinit/package.py]: You don't need to do use the ending range value if you just use `when="@9"`, which will mean all `9.x.y` (see docs at https://spack.readthedocs.io/en/latest/packaging_guide.html#version-ranges).

**Review by tldahlgren — DISMISSED** [review]

File: var/spack/repos/builtin/packages/abinit/package.py (hunk 1 above)
Comment (downloadico) [reply var/spack/repos/builtin/packages/abinit/package.py]: Thanks!

**Review by tldahlgren — APPROVED** [review]
Thanks.

**Pull request merged**